### ISSUE TITLE
gRPC Ruby on Mac requires Ruby > 2.0

### ISSUE THREAD

**Issue by jcanizales** [grpc/grpc#3798]
https://github.com/grpc/grpc/blob/master/examples/ruby/README.md says the requirement is Ruby 2.x, yet with version `ruby 2.0.0p481 (2014-05-08 revision 45883) [universal.x86_64-darwin14]` the extension can't be compiled (when installed via homebrew):

```
rb_call.c:142:5: error: excess elements in struct initializer [-Werror]
    0};
    ^
rb_call.c:156:5: error: excess elements in struct initializer [-Werror]
    0};
    ^
```

Inspecting the source code one can see the structure in question isn't gRPC-specific, rather part of `ruby.h`. Following the instructions in that README page to upgrade Ruby (getting version `ruby 2.2.1p85 (2015-02-26 revision 49769) [x86_64-darwin14]`) solves the issue.

**Comment by jcanizales** [comment]
A nit in the instructions to upgrade Ruby is that they say

> you can use RVM to use Ruby 2.x for testing without upgrading the version of Ruby on your whole system

leaving one looking for instructions to upgrade the version on the whole system (no temporary testing). Turns out RVM just works as well, but the time being confused was already spent :)

**Comment by tbetbetbe** [comment]
Closing, this has been replaced by #4020

**Comment by jcanizales** [comment]
Providing (or linking to) instructions to upgrade the Ruby version on the whole system, instead of just temporary testing, would still be nice.

**Comment by tbetbetbe** [comment]
- Please read the docs on [RVM](http://rvm.io).  RVM is not used for temporary testing, it's about managing your an entire ruby environment.  E.g, you could set up cocoapods in an rvm environment and alway switch to that environment whenever you wanted to use cocoapods.  This is better than installing cocoapads to the system; even without this issue with gRPC, it means that always have a separate problem of ensuring that the dependencies of any other ruby package that is installed are consistent with those used by cocoapod's.  That's the problem the [RVM](http://rvm.io) or [rbenv](https://github.com/sstephenson/rbenv) solves.

- The reason #4020 replaces this bug, is because it's being resolved by #4120, which ensures that you don't need upgrade the system ruby anyway; you can just use it.
